src/swupdd-main.c

```diff
@@ -22,751 +18,765 @@
-                                   _on_name_acquired,
-                                   _on_name_lost,
-                                   &dstate,
-                                   NULL);
-
-    g_main_loop_run (main_loop);
-
-    _destroy_interface (&dstate);
-    g_bus_unown_name (dstate.busid);
+	sd_bus *bus = context->bus;
+	bool exiting = false;
+	int r, code;
+
+	for (;;) {
+		r = sd_event_get_state(event);
+		if (r < 0) {
+			ERR("Failed to get event loop's state: %s", strerror(-r));
+			return r;
+		}
+		if (r == SD_EVENT_FINISHED) {
+			break;
+		}
+
+		r = sd_event_run(event, exiting ? (uint64_t) -1 : (uint64_t) TIMEOUT_EXIT_SEC * 1000000);
+		if (r < 0) {
+			ERR("Failed to run event loop: %s", strerror(-r));
+			return r;
+		}
+
+		if (!context->method && r == 0 && !exiting) {
+			r = sd_bus_try_close(bus);
+			if (r == -EBUSY) {
+				continue;
+			}
+                        /* Fallback for dbus1 connections: we
+                         * unregister the name and wait for the
+                         * response to come through for it */
+                        if (r == -EOPNOTSUPP) {
+				sd_notify(false, "STOPPING=1");
+
+				/* We unregister the name here and then wait for the
+				 * NameOwnerChanged signal for this event to arrive before we
+				 * quit. We do this in order to make sure that any queued
+				 * requests are still processed before we really exit. */
+
+				char *match = NULL;
+				const char *unique;
+				r = sd_bus_get_unique_name(bus, &unique);
+				if (r < 0) {
+					ERR("Failed to get unique D-Bus name: %s", strerror(-r));
+					return r;
+				}
+				r = asprintf(&match,
+					     "sender='org.freedesktop.DBus',"
+					     "type='signal',"
+					     "interface='org.freedesktop.DBus',"
+					     "member='NameOwnerChanged',"
+					     "path='/org/freedesktop/DBus',"
+					     "arg0='org.O1.swupdd.Client',"
+					     "arg1='%s',"
+					     "arg2=''", unique);
+				if (r < 0) {
+					ERR("Not enough memory to allocate string");
+					return r;
+				}
+
+				r = sd_bus_add_match(bus, NULL, match, on_name_owner_change, event);
+				if (r < 0) {
+					ERR("Failed to add signal listener: %s", strerror(-r));
+					free(match);
+					return r;
+				}
+
+				r = sd_bus_release_name(bus, "org.O1.swupdd.Client");
+				if (r < 0) {
+					ERR("Failed to release service name: %s", strerror(-r));
+					free(match);
+					return r;
+				}
+
+				exiting = true;
+				free(match);
+				continue;
+			}
+
+			if (r < 0) {
+				ERR("Failed to close bus: %s", strerror(-r));
+				return r;
+			}
+			sd_event_exit(event, 0);
+			break;
+		}
+	}
+
+        r = sd_event_get_exit_code(event, &code);
+        if (r < 0) {
+                return r;
+	}
+
+        return code;
+}
 
-    g_source_remove (sigid_int);
-    g_source_remove (sigid_term);
+static const sd_bus_vtable swupdd_vtable[] = {
+	SD_BUS_VTABLE_START(0),
+	SD_BUS_METHOD("checkUpdate", "a{sv}s", "b", method_check_update, 0),
```

Comment:
I was told DBus name schema is to use a Capital letter in methods and signals, so `CheckUpdate` and so on.